src/test/java/uk/gov/hmcts/reform/sscs/callback/handlers/RoboticsCallbackHandlerTest.java

```diff
@@ -226,4 +227,43 @@ public void givenARoboticsRequestFromReviewConfidentialityRequestAndStateIsRespo
         assertFalse(handler.canHandle(SUBMITTED, callback));
     }
 
+    @Test
+    public void givenARoboticsRequestFromDwpRaiseExceptionAndStateIsWithDwp_thenSendCaseToRobotics() {
+        handler = new RoboticsCallbackHandler(roboticsService, ccdService, idamService, regionalProcessingCenterService, true);
+
+        CaseDetails<SscsCaseData> caseDetails = getCaseDetails(WITH_DWP, READY_TO_LIST.getId());
+        Callback<SscsCaseData> callback = new Callback<>(caseDetails, Optional.empty(), EventType.DWP_RAISE_EXCEPTION, false);
+        caseDetails.getCaseData().setIsProgressingViaGaps("Yes");
+
+        handler.handle(SUBMITTED, callback);
+
+        verify(roboticsService).sendCaseToRobotics(any());
+
+        assertEquals(LocalDate.now().toString(), callback.getCaseDetails().getCaseData().getDateCaseSentToGaps());
+
+        ArgumentCaptor<String> capture = ArgumentCaptor.forClass(String.class);
+        verify(ccdService).updateCase(any(), any(), capture.capture(), any(), any(), any());
+
+        assertEquals(NOT_LISTABLE.getCcdType(), capture.getValue());
+    }
+
+    @Test
+    public void givenARoboticsRequestFromDwpRaiseExceptionAndStateIsWithDwpUcNotEnabled_thenCantHandle() {
+        CaseDetails<SscsCaseData> caseDetails = getCaseDetails(WITH_DWP, READY_TO_LIST.getId());
+        Callback<SscsCaseData> callback = new Callback<>(caseDetails, Optional.empty(), EventType.DWP_RAISE_EXCEPTION, false);
+        caseDetails.getCaseData().setIsProgressingViaGaps("Yes");
+
+        assertFalse(handler.canHandle(SUBMITTED, callback));
+    }
+
+    @Test
+    public void givenACcdExceptionAndStateIsWithDwpUcEnabled_thenNoExceptionThrown() {
+        handler = new RoboticsCallbackHandler(roboticsService, ccdService, idamService, regionalProcessingCenterService, true);
+        CaseDetails<SscsCaseData> caseDetails = getCaseDetails(WITH_DWP, READY_TO_LIST.getId());
+        Callback<SscsCaseData> callback = new Callback<>(caseDetails, Optional.empty(), EventType.DWP_RAISE_EXCEPTION, false);
+
+        ccdService = null;
```

Comment:
I don't like this test. 
It's better to verify no calls to ccdService, than to do this. This is because at least we are asserting something rather than the method doesn't throw an exception.